x-pack/plugins/security/public/account_management/user_profile/user_profile_api_client.ts

```diff
@@ -51,8 +73,20 @@ export class UserProfileAPIClient implements UserProfileAPIClientType {
       .then((response) => {
         const data = response?.data ?? {};
         const updated = merge(this._userProfile$.getValue(), data);
+
         this._userProfile$.next(updated);
+        this._enabled$.next(true);
```

Comment:
I think `userProfile$` having a profile available would be sufficient for the downstream checks, but maybe Im not understanding fully.

enabled$ and userProfileLoaded$ seem redundant

Reply:
I like code to be explicit instead of implicit. It helps in debugging issues. Imagine the following scenario:

1. Class is instantiated, `userProfile$` is `null`
2. My code waits for `userProfile$` value to be **not** `null` so I know that it was loaded
3. The server respond with `null` and update the Observable to ... `null` 
4. we got an issue

You gave `null` an extra meaning, a business rule, and now the server can't use that value without breaking someone downstream.